src/pip/_internal/network/auth.py

```diff
@@ -23,51 +28,153 @@
 
 logger = getLogger(__name__)
 
-Credentials = Tuple[str, str, str]
+KEYRING_DISABLED = False
 
-try:
-    import keyring
-except ImportError:
-    keyring = None  # type: ignore[assignment]
-except Exception as exc:
-    logger.warning(
-        "Keyring is skipped due to an exception: %s",
-        str(exc),
-    )
```

Comment:
Perhaps we retain this message if `import keyring` fails? It may not be the most intuitive if we silently fallback to `keyring` on PATH when this happens, so we should inform the user.

Reply:
Agreed. Maybe re-word it slightly: "Installed copy of keyring fails with exception %s, trying to find a keyring executable as a fallback".

Reply:
Do we think we should try to fallback at all in the event of an unexpected exception on import?

Reply:
I think we should either fallback or disable keyring; pip should at least maintain some working state instead of just crashing.